traitlets/config/configurable.py

```diff
@@ -187,6 +187,10 @@ def _config_changed(self, change):
 
     def update_config(self, config):
         """Update config and load the new values"""
+        # make a copy of our config prior to loading to protect IPython,
```

Comment:
Rather than this being an IPython-specific hack, which is what the comments seems to imply, we should indicate whether we mean these semantics to be how the system is designed to work, and in that case IPython would be making a proper, supported assumption.